### PR TITLE
Add a stock Ord instance for AlexPosn

### LINKED ISSUE #233: Deriving Ord for AlexPosn
I'm trying to use Alex with Megaparsec, whose [Stream](https://hackage.haskell.org/package/megaparsec-9.3.0/docs/Text-Megaparsec-Stream.html#t:Stream) typeclass requires an Ord constraint on the elements of a list or sequence of tokens.

As I pre-lex my source input with Alex and turn it into a Seq of Ranged Tokens making use of AlexPosn for the position, it is the last thing that does not have an Ord instance.

I can provide a patch.

### PR DESCRIPTION
**Pull request by Kleidukos**
Close #233

This PR adds an Ord instance to the generated and the internal AlexPosn struct.

### CODE REVIEW

**Review by andreasabel — APPROVED** [review]

**Comment by andreasabel** [comment]
@Ericson2314 :
This change will break compilation of project that add an orphan `instance Ord AlexPosn`, so I want a second opinion, e.g. what versioning concerns (3.2.8 vs. 3.3).

**Comment by andreasabel** [comment]
https://hackage-search.serokell.io/?q=instance+Ord+AlexPosn gives the following conflicts:
Screenshot 2023-05-14 at 19 18 45
- https://github.com/TokTok/hs-cimple
- https://github.com/vmchale/jacinda
- https://github.com/vmchale/kempe

**Comment by andreasabel** [comment]
Resolved the conflict; planning to release this as 3.3.0.

File: src/ParseMonad.hs
Comment (andreasabel) [inline src/ParseMonad.hs:67]: Sorry, now that I have another look, I see that you only add this to `alex` itself not to the _lexers_ that `alex` generates.

What is the point of this?

**Review by andreasabel — CHANGES_REQUESTED** [review]
Please clarify what this PR wants to achieve and argue why it achieves it.  
Also a test please.

File: src/ParseMonad.hs
Comment (Kleidukos) [reply src/ParseMonad.hs:67]: ah, well-spotted! fixing this :)

**Comment by Kleidukos** [comment]
@andreasabel I'm slightly lost wrt the test suite. Despite having changed the wrapper, I don't think the test suite actually failed on the content change of the generated lexer, despite said content having changed in my tree.

```
❯ grep -R "deriving (Eq, Show, Ord)"
basic_typeclass.n.hs:        deriving (Eq, Show, Ord)
basic_typeclass.g.hs:        deriving (Eq, Show, Ord)
[…]
```

**Review by andreasabel — APPROVED** [review]
Thanks!
I added a test for the availability of the `instance Ord AlexPosn`.

**Comment by andreasabel** [comment]
> I don't think the test suite actually failed on the content change of the generated lexer,

Maybe you expected a "golden value" test, which does not exist here.  The testsuite just generates lexers from `.x` files, I think it does not even try to run them.  The generated lexers are not checked in as golden values.

**Comment by Kleidukos** [comment]
ooooooh I see! Thank you :)

**Comment by Kleidukos** [comment]
@andreasabel understood, I'll merge it then.

**Comment by andreasabel** [comment]
> @andreasabel understood, I'll merge it then.

Maybe the final "hit the button" should be priviledge of the maintainers (stern look).
Or do you want to apply for co-maintainership?

**Comment by Kleidukos** [comment]
My apologies, I was operating under a "take responsibility of your contributions" model.

I am a very poor user of Alex right now, but perhaps after a few other contributions I would certainly consider helping out with maintenance. :)

**Comment by andreasabel** [comment]
I would have reorganized the commits a bit so that testcase is in the same commit with the change, but so be it now.

> perhaps after a few other contributions I would certainly consider helping out with maintenance. :)

Contributions are very welcome!  
I think it is also nice if maintainers have sparring partners, so that one can get a second option.

**Comment by Ericson2314** [comment]
@andreasabel I am very sorry I did not respond to this in time.

**Comment by Ericson2314** [comment]
Looking at it after the fact, the change and the version number bump seem good to me. The state of the version numbers on Alex and Happy is a little weird, but hopefully we can consistently get back to 4 digit ones so the interpretation of the PVP is most obvious.

**Comment by andreasabel** [comment]
@Ericson2314 
> @andreasabel I am very sorry I did not respond to this in time.

No worries!

> Looking at it after the fact, the change and the version number bump seem good to me. The state of the version numbers on Alex and Happy is a little weird, but hopefully we can consistently get back to 4 digit ones so the interpretation of the PVP is most obvious.

We can release this as 3.3, 3.3.0, or 3.3.0.0.  Technically, this is a strictly increasing version sequence, but non-experts (like my former self) would consider these equal.

Historically, Alex had both x.y (2.2, 2.3, 3.0) and x.y.0 (2.1.0, 3.1.0, 3.2.0) with x.y.0 dominant recently, so I went for 3.3.0.
However, we could continue our 4-digit streak and switch to exactly 4 digits _always_, so 3.3.0.0 would be next.

I take your statement as a vote for 3.3.0.0, @Ericson2314, is this your intention?

**Comment by andreasabel** [comment]
Release candidate for 3.3.0.0 at:
- https://github.com/haskell/alex/pull/239

**Pull request merged**
